src/core/border_router/routing_manager.cpp

```diff
@@ -204,7 +213,57 @@ void RoutingManager::GenerateOmrPrefix(void)
     LogInfo("Generated OMR prefix: %s", mLocalOmrPrefix.ToString().AsCString());
 }
 
+const char    RoutingManager::kWellKnownIpv4OnlyName[]     = "ipv4only.arpa";
+const uint8_t RoutingManager::kWellKnownIpv4OnlyAddress1[] = {192, 0, 0, 170};
+const uint8_t RoutingManager::kWellKnownIpv4OnlyAddress2[] = {192, 0, 0, 171};
+
 #if OPENTHREAD_CONFIG_BORDER_ROUTING_NAT64_ENABLE
+void RoutingManager::RequestInfraIfNat64Prefix(void)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(IsInitialized() && mInfraIf.IsRunning(), error = kErrorInvalidState);
+
+    LogInfo("Requesting infrastructure NAT64 prefix from %s on %s", kWellKnownIpv4OnlyName,
+            mInfraIf.ToString().AsCString());
+    error = mInfraIf.RequestHostAddress(kWellKnownIpv4OnlyName);
+
+exit:
+    if (error != kErrorNone)
+    {
+        LogWarn("Failed to request upstream NAT64 prefix on %s: %s", mInfraIf.ToString().AsCString(),
+                ErrorToString(error));
+    }
+}
+
+void RoutingManager::UpdateInfraIfNat64Prefix(const Ip6::Prefix &aPrefix)
+{
+    if (!mIsRunning)
+    {
+        mInfraIfNat64Prefix = aPrefix;
+        LogInfo("Get infrastructure NAT64 prefix %s", mInfraIfNat64Prefix.ToString().AsCString());
+        ExitNow();
+    }
+
+    if ((!aPrefix.IsValidNat64() && !mInfraIfNat64Prefix.IsValidNat64()) ||
+        (aPrefix.IsValidNat64() && aPrefix == mInfraIfNat64Prefix))
+    {
+        LogInfo("Infrastructure NAT64 prefix %s doesn't change",
+                mInfraIfNat64Prefix.IsValidNat64() ? mInfraIfNat64Prefix.ToString().AsCString() : "none");
+    }
+    else
+    {
+        LogNote("Infrastructure NAT64 prefix changed from %s to %s",
+                mInfraIfNat64Prefix.IsValidNat64() ? mInfraIfNat64Prefix.ToString().AsCString() : "none",
+                aPrefix.IsValidNat64() ? aPrefix.ToString().AsCString() : "none");
+        EvaluateNat64Prefix(aPrefix);
+        mInfraIfNat64Prefix = aPrefix;
```

Comment:
```suggestion
        mInfraIfNat64Prefix = aPrefix;
        StartRoutingPolicyEvaluationJitter();
```

I think  it's better to trigger routing policy evaluation with a jitter

Reply:
`mInfraIfNat64Prefix` shall be updated with the new infra NAT64 prefix after the evaluation, otherwise the legacy prefix will be wiped out and fail the evaluation.

Reply:
Can we save the old `mInfraIfNat64Prefix` and then start the evaluation jitter?

Reply:
Replaced the `mNat64PrefixStatus` with `mAdvertisedNat64Prefix` to both indicate the advertising status and save the old AIL NAT64 prefix. Only the AIL NAT64 prefix that was advertised would be saved and used in next evaluation. Triggering the evaluation with a jitter works in this way.